docs/proxmox.md

```diff
@@ -0,0 +1,94 @@
+# Proxmox Virtual Environment
+
+## State of the implementation
+
+Support for Proxmox as a provider in the machine-controller is currently just a technical demo. It
+is possible to create MachineDeployments using manually created VM templates as demonstrated below.
+In this example the VM template is using local storage, which is why this template can only be
+cloned on the same node it is located at.
+
+## Prerequisites
+
+### Authentication
+
+For authentication the following data is needed:
+
+- `user_id` is expected to be in the form `USER@REALM!TOKENID`
+- `token` is just the UUID you get when initially creating the token
+
+See also:
+* https://pve.proxmox.com/wiki/User_Management#pveum_tokens
+* https://pve.proxmox.com/wiki/Proxmox_VE_API#API_Tokens
+
+#### User Privileges
+
+For the provider to properly function the user needs an API token with the following privileges:
+
+* `Datastore.AllocateSpace`
+* `Pool.Allocate`
+* `Pool.Audit`
+* `VM.Allocate`
+* `VM.Audit`
+* `VM.Clone`
+* `VM.Config.CDROM`
+* `VM.Config.CPU`
+* `VM.Config.Cloudinit`
+* `VM.Config.Disk`
+* `VM.Config.HWType`
+* `VM.Config.Memory`
+* `VM.Config.Network`
+* `VM.Config.Options`
+* `VM.Monitor`
+* `Sys.Audit`
+* `Sys.Console`
+
+### Cloud-Init enabled VM Templates
+
+Although it is possible to upload Cloud-Init images in Proxmox VE and create VM disks directly from
+these images via CLI tools on the nodes directly, there is no API endpoint yet to provide this
+functionality externally. That's why the `proxmox` provider assumes there are VM templates in place
+to clone new machines from.
+
+Proxmox recommends using either ready-to-use Cloud-Init images provided by many Linux distributions
+(mostly designed for OpenStack) or to prepare the images yourself as you have full control over
+what's in these images.
+
+For VM templates to be available on all nodes, they need to be added to the `ha-manager`.
+
+Example for creating a VM template:
```

Comment:
tiny nitpick: I think a link to the qm tool would be useful. Maybe it could be easily integrated like this

```suggestion
Example for creating a VM template using [qm](https://pve.proxmox.com/pve-docs/qm.1.html):
```